ngi_pipeline/utils/charon.py

```diff
@@ -0,0 +1,45 @@
+import collections
+import re
+
+from ngi_pipeline.database.classes import CharonSession, CharonError
+from ngi_pipeline.log.loggers import minimal_logger
+
+LOG = minimal_logger(__name__)
+
+def find_projects_from_samples(sample_list):
+    """Given a list of samples, attempts to determine
+    which projects they belong to using Charon records.
+
+    :param list sample_list: A list of the samples for which to find projects
+
+    :returns: a dict of {project_id: set(samples)}
+    :rtype: dict of sets
+    """
+    STHLM_SAMPLE_RE = re.compile(r'(P\d{4})_')
+    projects_dict = collections.defaultdict(set)
+    samples_by_project_id = {}
+    no_owners_found = set()
+    multiple_owners_found = set()
+    charon_session = CharonSession()
+
+    for sample_name in sample_list:
+        # First see if we can just parse out the project id from the sample name
+        m = STHLM_SAMPLE_RE.match(sample_name)
+        if m:
+            project_id = m.groups()[0]
+            projects_dict[project_id].add(sample_name)
+        else:
+            # Otherwise check all the projects for matching samples
+            owner_projects_list = charon_session.sample_get_projects(sample_name).get('projects')
```

Comment:
@Galithil is this correct? The structure of the JSON string returned, I mean. I can test it when you deploy the Charon PR but I just took a guess here.